### PR TITLE
Onboard ProviderHub SDK

### PR DESCRIPTION
**Pull request by wwendyc**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [x] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [x] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running generate.ps1. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version,

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

File: sdk/providerhub/Microsoft.Azure.Management.ProviderHub/src/Microsoft.Azure.Management.ProviderHub.csproj [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <!-- Please do not move/edit code below this line -->
+  <Import Project="$(RepoEngPath)/mgmt/AzSdk.reference.props" />
+  <!-- Please do not move/edit code below this line -->
+
+  <PropertyGroup>
+    <PackageId>Microsoft.Azure.Management.ProviderHub</PackageId>
+    <Description>Microsoft Azure Management Library for ProviderHub operations.</Description>
+    <Version>1.0.0</Version>
```
Comment (markcowl) [inline sdk/providerhub/Microsoft.Azure.Management.ProviderHub/src/Microsoft.Azure.Management.ProviderHub.csproj:9]: Your fist version should likely be a preview.  version 1.0.0-beta.1  would be the correct version number for your first preview

File: sdk/providerhub/Microsoft.Azure.Management.ProviderHub/src/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Reflection;
+using System.Resources;
+
+[assembly: AssemblyTitle("Microsoft Azure ProviderHub Management Library")]
+[assembly: AssemblyDescription("Provides management functionality for Microsoft Azure ProviderHub Resources.")]
+
+[assembly: AssemblyVersion("1.0.0.0")]
+[assembly: AssemblyFileVersion("1.0.0.0")]
```
Comment (markcowl) [inline sdk/providerhub/Microsoft.Azure.Management.ProviderHub/src/Properties/AssemblyInfo.cs:11]: AssemblyFileversion should be 1.0.0.1 if you use the beta version

**Review by markcowl — CHANGES_REQUESTED** [review]
- Please include a link to your swagger PR in the comments.
- The ci issue looks like there is an issue with a test recording - usually this indicates a problem with client setup  (client does not have the recording interceptor in its pipeline)

File: sdk/support/Microsoft.Azure.Management.Support/tests/Helpers/RecordedDelegatingHandler.cs [hunk 3]
```diff
@@ -1,90 +0,0 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (markcowl) [inline sdk/support/Microsoft.Azure.Management.Support/tests/Helpers/RecordedDelegatingHandler.cs:1]: I don't think this change should be included

File: sdk/support/Microsoft.Azure.Management.Support/tests/Helpers/Util.cs [hunk 4]
```diff
@@ -1,45 +0,0 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (markcowl) [inline sdk/support/Microsoft.Azure.Management.Support/tests/Helpers/Util.cs:1]: Or this one

**Review by markcowl — CHANGES_REQUESTED** [review]

**Comment by markcowl** [comment]
@wwendyc Please remove changes to the support management library tests, as indicated above.

File: sdk/providerhub/Microsoft.Azure.Management.ProviderHub/tests/Microsoft.Azure.Management.ProviderHub.Tests.csproj [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  </ItemGroup>
+
+  <ItemGroup>
+    <None Update="SessionRecords\**\*.json">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </None>
+  </ItemGroup>
+
+
+  <ItemGroup>
```
Comment (markcowl) [inline sdk/providerhub/Microsoft.Azure.Management.ProviderHub/tests/Microsoft.Azure.Management.ProviderHub.Tests.csproj:27]: don't need this

**Comment by wwendyc** [comment]
Swagger link: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/providerhub/resource-manager/Microsoft.ProviderHub/stable/2020-11-20/providerhub.json

File: sdk/providerhub/Microsoft.Azure.Management.ProviderHub/tests/Microsoft.Azure.Management.ProviderHub.Tests.csproj (hunk 5 above)
Comment (wwendyc) [reply sdk/providerhub/Microsoft.Azure.Management.ProviderHub/tests/Microsoft.Azure.Management.ProviderHub.Tests.csproj:27]: The lines highlighted don't seem to have any service reference, could you please clarify which one?

**Review by markcowl — APPROVED** [review]

**Pull request merged**